pkg/controller/metrics/metricscontroller.go

```diff
@@ -136,45 +143,54 @@ func (c *controller) GetLatestMetrics(opt *pb.GetMetricsOpts) (*[]model.MetricSp
 		if err != nil {
 			log.Error(err)
 		}
+		return metrics, err
 
 	}
-	return &metrics, err
+	//no response
+	return nil, err
 }
 
-func (c *controller) GetInstantMetrics(opt *pb.GetMetricsOpts) (*[]model.MetricSpec, error) {
+func (c *controller) GetInstantMetrics(opt *pb.GetMetricsOpts) ([]*model.MetricSpec, error) {
 
-	var metrics []model.MetricSpec
 	// make a call to Prometheus, convert the response to our format, return
 	response, err := http.Get("http://localhost:9090/api/v1/query?query=" + opt.MetricName + "&time=" + opt.StartTime)
```

Comment:
Please make this URL configurable.

Reply:
We have plan to handle all such reference to a configuration file . Back log is already raised. https://github.com/opensds/opensds/issues/691